example.html

```diff
@@ -1,43 +1,73 @@
-<html>
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>Ably Presence</title>
+  </head>
   <body>
     <h1>Presence example</h1>
 
-    <h2>Live-updated presence set:</h2>
+    <h2>Present in the "chatroom" channel:</h2>
     <ul id="presence-set"></ul>
   </body>
+  <!-- Import the Ably client library-->
+  <script src="https://cdn.ably.com/lib/ably.min-1.js"></script>
 
-  <!-- Include the latest Ably Library  -->
-  <script src="https://cdn.ably.io/lib/ably.min-1.js"></script>
-
-  <!-- Instance the Ably library  -->
-  <script type="text/javascript">
-    alert("You need to add your API key"); /* REMOVE THIS */
-    var realtime = new Ably.Realtime({
-      key: "INSERT-YOUR-API-KEY-HERE", /* ADD YOUR API KEY HERE */
-      clientId: "my-client-id", /* This is who you will appear as in the presence set */
-      closeOnUnload: true // See https://support.ably.io/solution/articles/3000059875-why-don-t-presence-members-leave-as-soon-as-i-close-a-tab-
+  <script>
+    const apiKey = "YOUR_ABLY_API_KEY"; // Replace with your API key
+    const randomId =
+      Math.random().toString(36).substring(2, 10) +
+      Math.random().toString(36).substring(2, 10);
```

Comment:
a simple alternative 

```suggestion
    const randomId = Math.random().toString(36).slice(-10);
```

Reply:
Nice